src/resource/solidDataset.test.ts

```diff
@@ -2017,6 +2017,146 @@ describe("createContainerAt", () => {
+        );
+        mockDataset = setThing(mockDataset, additionThing);
+      });
+
+      const resourceInfo: WithResourceInfo["internal_resourceInfo"] = {
+        sourceIri: fromUrl,
+        isRawData: false,
+      };
+
+      return {
+        ...mockDataset,
+        internal_changeLog: changeLog,
+        internal_resourceInfo: resourceInfo,
+      };
+    }
+
+    it("throws if container SolidDataset has updates", async () => {
+      const mockFetch = jest
+        .fn(window.fetch)
+        .mockReturnValue(
+          Promise.resolve(
+            mockResponse(undefined, { url: "https://arbitrary.pod/container/" })
+          )
+        );
+
+      const mockDataset = getMockUpdatedDataset(
+        {
+          additions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some.vocab/subject"),
+              DataFactory.namedNode("https://some.vocab/predicate"),
+              DataFactory.namedNode("https://some.vocab/object"),
+              undefined
+            ),
+          ],
+          deletions: [],
+        },
+        "https://arbitrary.pod/container/"
+      );
+
+      await expect(
+        createContainerAt(
+          "https://arbitrary.pod/container/",
+          {
+            fetch: mockFetch,
+          },
+          mockDataset
+        )
+      ).rejects.toThrow("dataset that contains updates");
+    });
+
+    it("returns an non-empty SolidDataset if one is provided", async () => {
+      const mockFetch = jest
+        .fn(window.fetch)
+        .mockReturnValue(
+          Promise.resolve(
+            mockResponse(undefined, { url: "https://arbitrary.pod/container/" })
+          )
+        );
+
+      const mockThing = addUrl(
+        createThing({ url: "https://arbitrary.vocab/subject" }),
+        "https://arbitrary.vocab/predicate",
+        "https://arbitrary.vocab/object"
+      );
+      const mockDataset = setThing(createSolidDataset(), mockThing);
+
+      const solidDataset = await createContainerAt(
+        "https://arbitrary.pod/container/",
+        {
+          fetch: mockFetch,
+        },
+        mockDataset
+      );
+
+      expect(mockFetch.mock.calls).toHaveLength(1);
+      expect(mockFetch.mock.calls[0][0]).toEqual(
+        "https://arbitrary.pod/container/"
+      );
+      expect(mockFetch.mock.calls[0][1]?.method).toBe("PUT");
+      expect(
+        (mockFetch.mock.calls[0][1]?.headers as Record<string, string>)[
+          "Content-Type"
+        ]
+      ).toBe("text/turtle");
+      expect(
+        (mockFetch.mock.calls[0][1]?.headers as Record<string, string>)["Link"]
+      ).toBe('<http://www.w3.org/ns/ldp#BasicContainer>; rel="type"');
+      expect((mockFetch.mock.calls[0][1]?.body as string).trim()).toBe(
+        "<https://arbitrary.vocab/subject> <https://arbitrary.vocab/predicate> <https://arbitrary.vocab/object>."
+      );
+    });
+
+    it("reports HTTP error when an non-empty SolidDataset provided", async () => {
```

Comment:
I think the suggested refactoring makes this test redundant, as one branch has been removed from the code.